### PR TITLE
Dependency updates and deprecation fixes

### PR DESCRIPTION
**Pull request by TacoTheDank**
I really hope you don't mind big pull requests. Each commit is its own little part.

### 1st commit
- Deleted unnecessary files (like .idea, and unused test files)

### 2nd commit
- Updated gradle wrapper to 6.6.1 using git bash

### 3rd commit
- Rearranged gradles for easier reading
- Added Java 1.8 support
- Grouped dependencies
- Updated Junit 4.12 -> 4.13
- Updated Studio 4.0.0 -> 4.0.1
- Updated play-publisher 1.2.0 -> 1.2.2 ([changelog](https://github.com/Triple-T/gradle-play-publisher/releases?after=2.0.0))

### 4th commit
- Migrated to AndroidX

### 5th commit
- Optimized imports

### 6th commit
From their initial AndroidX versions:
- AppCompat 1.0.2 -> 1.2.0 ([changelog](https://developer.android.com/jetpack/androidx/releases/appcompat))
- SwipeRefreshLayout 1.0.0 -> 1.1.0 ([changelog](https://developer.android.com/jetpack/androidx/releases/swiperefreshlayout))
- Material 1.0.0 -> 1.2.1 ([changelog](https://github.com/material-components/material-components-android/releases))
- Deleted unused libraries

### 7th commit
- Updated android-file-chooser 1.1.5 -> 1.2.0-final ([changelog](https://github.com/hedzr/android-file-chooser/blob/master/CHANGELOG))
- Updated android-async-http 1.4.9 -> 1.4.11 ([changelog](https://github.com/android-async-http/android-async-http/blob/master/CHANGELOG.md))
- Updated osmdroid-android 6.0.2 -> 6.1.8 ([changelog](https://github.com/osmdroid/osmdroid/releases))
- Updated mockito-core 3.2.4 -> 3.5.10 ([changelog](https://github.com/mockito/mockito/releases))

### 8th commit
- Fix a random deprecation

### 9th commit
- Fixed spelling of platfom-tools to platform-tools
- Updated the tools and stuff
- Updated the licenses

### CODE REVIEW

**Comment by TacoTheDank** [comment]
Any updates?

**Comment by TacoTheDank** [comment]
@thuryn Sorry for mention, don't know if you got this.

**Comment by thuryn** [comment]
sorry, I havn't noticed this (and all other PRs). I'll check it.

**Comment by iillyyaa2033** [comment]
@thuryn any updates?

**Comment by TacoTheDank** [comment]
I'll have to fix it first, I suppose.

**Comment by TacoTheDank** [comment]
@thuryn I decided to shrink the PR a little bit; I'll save the code optimizing for another PR.

Anyway, please review. Thanks :)

**Comment by thuryn** [comment]
I would like to exclude this commit "Adjust codestyle to an .editorconfig". Is this possible?

**Comment by TacoTheDank** [comment]
@thuryn Yeah, I'll do that for you.

**Comment by TacoTheDank** [comment]
@thuryn Done. 👍

**Review by thuryn — APPROVED** [review]

**Comment by thuryn** [comment]
Thank you for you time, your work and a patience.

File: app/src/main/java/org/thosp/yourlocalweather/MainActivity.java [hunk 1]
```diff
@@ -666,7 +658,7 @@ public boolean checkPermissionsSettingsAndShowAlert() {
                 return true;
             }
             settingsAlert.setMessage(notificationMessage.toString());
-            final String[] permissionsArray = permissions.toArray(new String[permissions.size()]);
+            final String[] permissionsArray = permissions.toArray(new String[0]);
```
[the remark this answers is not in the document]
Comment (TacoTheDank) [reply app/src/main/java/org/thosp/yourlocalweather/MainActivity.java]: This is recommended in newer Java versions, as it is now more efficient.

**Pull request merged**